File: ServerApp/ServerApp/GameState/Chest.cs
namespace ServerApp.GameState
{
    public class Chest : IInteractiveObject
    {
        public string Name {get;}
        public int goldContent {get; set;}

        public Chest() {
            Name = "Chest";
            goldContent = 100;
        }
        
        public void interact(IGameState gameState)
        {
            if(goldContent != 0) {
                gameState.Player.AddGold(goldContent);
                goldContent = 0;
            }
        }
    }
}

File: clientApp/src/GameState/Chest.ts
import { IInteractiveObject } from './IInteractiveObject';

export class Chest implements IInteractiveObject {
    public name: string;
    public goldContent: number;

    constructor(name: string, goldContent: number) {
        this.name  = name;
        this.goldContent = goldContent;
    }
}
